Model/Service/Category/CategoryMappingService.php

```diff
@@ -0,0 +1,140 @@
+ *
+ */
+
+namespace Nosto\Cmp\Model\Service\Category;
+
+use Exception;
+use Magento\Catalog\Model\Category;
+use Magento\Catalog\Model\ResourceModel\Category\CollectionFactory;
+use Magento\Store\Model\Store;
+use Nosto\Cmp\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Logger\Logger;
+use Nosto\Tagging\Model\Service\Product\Category\DefaultCategoryService as CategoryBuilder;
+
+class CategoryMappingService implements CategoryMappingServiceInterface
+{
+
+    /** @var CollectionFactory */
+    private $collectionFactory;
+
+    /** @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var Logger */
+    private $logger;
+
+    /**
+     * @param CollectionFactory $collectionFactory
+     * @param CategoryBuilder $categoryBuilder
+     * @param NostoHelperData $nostoHelperData
+     * @param Logger $logger
+     */
+    public function __construct(
+        CollectionFactory $collectionFactory,
+        CategoryBuilder $categoryBuilder,
+        NostoHelperData $nostoHelperData,
+        Logger $logger
+    ) {
+        $this->collectionFactory = $collectionFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->nostoHelperData = $nostoHelperData;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     * @return string
+     */
+    public function getCategoryMapping(Store $store): string
+    {
+        $array = $this->getMagentoCategories($store);
+        $mapping = json_encode((object)$array, JSON_UNESCAPED_SLASHES);
+        if ($mapping == false) {
+            return '';
+        }
+        return $mapping;
+    }
+
+    /**
+     * @param Store $store
+     * @return array
+     * @suppress PhanTypeMismatchArgument
+     */
+    private function getMagentoCategories(Store $store)
+    {
+
+        $categoriesArray = [];
+
+        try {
+            $categories = $this->collectionFactory->create()
+                ->addAttributeToSelect('*')
+                ->addIsActiveFilter()
+                ->setStore($store);
+
+            if (!$this->nostoHelperData->isAllCategoriesMapEnabled($store)) {
+                $categories->addAttributeToFilter('include_in_menu', ['eq' => 1]);
+            }
+
+            /** @var Category $category $item */
+            foreach ($categories->getItems() as $category) {
+                $hashedCategoryString = $this->hashCategoryString(strtolower(
+                    $this->categoryBuilder->getCategory($category, $store)
+                ));
+                if ($hashedCategoryString) {
+                    $categoriesArray[$hashedCategoryString] = $category->getUrl();
+                }
+            }
+        } catch (Exception $e) {
+            $this->logger->exception($e);
+        }
+
+        return $categoriesArray;
+    }
+
+    /**
+     * @param String $categoryString
+     * @return string
+     */
+    private function hashCategoryString($categoryString)
```

Comment:
Should this be `static `?

Reply:
Why should it be? It's referenced within the class and it's not a utility.

Reply:
Does it rely on the state of the class? If not, then it should be static. https://stackoverflow.com/questions/48755/when-should-a-method-be-static